test/tests/functional/pbs_reservations.py

```diff
@@ -1123,3 +1123,409 @@ def test_reservations_with_expired_subjobs(self):
+        elif 'PBS_TZID' in os.environ:
+            tzone = os.environ['PBS_TZID']
+        else:
+            self.logger.info('Missing timezone, using America/Los_Angeles')
+            tzone = 'America/Los_Angeles'
+        # Submit a standing reservation to occur every other minute for a
+        #  total count of 2
+        start = int(time.time()) + 10
+        end = start + 60
+        a = {'Resource_List.select': '1:ncpus=4',
+             ATTR_resv_rrule: 'FREQ=MINUTELY;INTERVAL=2;COUNT=2',
+             ATTR_resv_timezone: tzone,
+             'reserve_start': start,
+             'reserve_end': end,
+             }
+        r = Reservation(TEST_USER, attrs=a)
+        rid = self.server.submit(r)
+        exp_attr = {'reserve_state': (MATCH_RE, "RESV_CONFIRMED|2")}
+        self.server.expect(RESV, exp_attr, id=rid)
+        rid_q = rid.split(".")[0]
+        # Submit a job-array within resrvation
+        j = Job(TEST_USER, attrs={ATTR_q: rid_q, ATTR_J: '1-5'})
+        j.set_sleep_time(40)
+        jid = self.server.submit(j)
+        subjid = []
+        for i in range(1, 5):
+            subjid.append(j.create_subjob_id(jid, i))
+        self.server.expect(JOB, {'job_state': 'Q',
+                                 'comment': (MATCH_RE,
+                                             'Queue not started')}, jid)
+        # Wait for standing reservation first instance to start
+        exp_attr = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, exp_attr, id=rid, offset=10)
+        self.server.expect(RESV, {'reserve_index': 1}, id=rid)
+        self.server.expect(JOB, {'job_state': 'B'}, jid)
+        self.server.expect(JOB, {'job_state=R': 4}, count=True,
+                           id=jid, extend='t')
+        self.server.expect(JOB, {'job_state=Q': 1}, count=True,
+                           id=jid, extend='t')
+        # Wait for standing reservation first instance to finished
+        self.logger.info(
+            'Waiting 60 seconds for first instance of reservation to Finish')
+        time.sleep(60)
+        exp_attr = {'reserve_state': (MATCH_RE, "RESV_CONFIRMED|2")}
+        self.server.expect(RESV, exp_attr, id=rid)
+
+        # Wait for standing reservation second instance to start
+        self.logger.info(
+            'Waiting 60 seconds for second  instance of reservation to start')
+        self.server.expect(RESV, {'reserve_index': 2}, offset=60, id=rid)
+        exp_attr = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, exp_attr, id=rid)
+        # Wait for reservations to be finished
+        msg = "Que;" + rid_q + ";deleted at request of pbs_server@"
+        self.server.log_match(msg, starttime=end, interval=20)
+
+        # Check for finished jobs by issuing the command qstat
+        self.server.expect(JOB, {'job_state': 'F', 'substate': '93'},
+                           extend='x', id=subjid[3], offset=65)
```

Comment:
Why is offset set to 65? since this is the 4th job, I assume this job would have ended with the previous reservation instance so ideally should be finished by now.